src/runtime/app.ts

```diff
@@ -101,6 +108,30 @@ export class App<
     return prom;
   }
 
+  createRoot<SubProps extends object>(
+    SubRoot: ComponentConstructor<SubProps, E>,
+    subProps: SubProps
+  ): SubRoot<SubProps, E> {
+    const node = this.makeNode(SubRoot, subProps);
+    this.subRoots.add(node);
+    return {
+      root: node,
+      mount: (target: HTMLElement | ShadowRoot, options?: MountOptions) => {
+        App.validateTarget(target);
+        if (this.dev) {
+          validateProps(SubRoot, subProps, { __owl__: { app: this } });
+        }
+        const prom = this.mountNode(node, target, options);
+        return prom;
```

Comment:
Shouldn't mount call createRoot?

Reply:
done (but not as elegant as one might expect, i had to remove something from the subroot sets. this could probably be improved in the future in owl 3. i think the App class is not designed properly, it should basically not have a mount method, only a createRoot.